### PR TITLE
Post on first training session

### PR DESCRIPTION
**Pull request by maelle**

### CODE REVIEW

**Comment by yabellini** [comment]
I added:

- What makes a good function.
-  Consistency on how to form names.
-  camelCase

**Comment by maelle** [comment]
@yabellini thank you!! I wonder whether we should add a caveat regarding camelCase, as we also know it's not the most popular one in the R community at the moment (I know I use snakecase).

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+- Felienne Hermans, in her book [The Programmer's Brain](https://www.manning.com/books/the-programmers-brain), advises choosing the concepts that go into the name, the words to say it, then putting them together. This approach in three steps is a good way to get unstuck.
+
+- Following the previous advice, names should be consistent across code base and _name molds_ are a very good tool for that. Name molds are patterns in which the elements of a name are combined, for example if you calculate the maximum value of crop yield, you need to agree if `maximun` will be `max` or `maximun` and if the word will be at the beginning or at the end of the variable name: should be `maxYield` or `yieldMax`?  By normalizing how to name things, our code will be more easy to read.     
+
+- “The greater the distance between a name’s declaration and its uses, the longer the name should be” (Andrew Gerrand). However, no matter, how close to defining a variable you use it, don't use a smart very short abbreviation.
+
+- There are several way to write variables names, camelCase style lead to higher accuracy when reading code (Dave Binkley, 2009) and is better for reading the code with screen readers.
```
Comment (yabellini) [inline content/blog/2024-02-20-stylish-code/index.Rmd:166]:
```suggestion
- There are several ways to write variable names. camelCase style leads to higher accuracy when reading code (Dave Binkley, 2009) and is better for reading the code with screen readers.  We know it is difficult to change the style of an existing project, but if you are in a situation where you can decide from scratch, then consider using Camel Case. 
```

Comment (yabellini) [reply content/blog/2024-02-20-stylish-code/index.Rmd:166]: Something like that?

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+It is important to know how to quickly navigate between functions in your IDE!
+In RStudio IDE, you can use Ctrl+click on the function name, or type its name in the search bar.
+
+A third way to shorten your code is to use existing functions from base R or add-on packages!
+For instance, to combine a list of default values with a list of custom values, you can use the [`modifyList()` function](https://masalmon.eu/2023/06/06/basic-patterns/#combine-a-list-of-default-values-with-a-list-of-custom-values).
+As with human languages, we learn more R words over time, by reading other people's code and having them read our code.
+A potential nice thing to do is to [share one's learnings](https://masalmon.eu/tags/useful-functions/).
+
+### Not too stained code: just the right amount of comments
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:116]: Not sure about the use of 'stained'...
```suggestion
### Not too wordy: just the right amount of comments
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+This part of the training was a shorter version of the R-hub blog post [Why comment your code as little (and as well) as possible](https://blog.r-hub.io/2023/01/26/code-comments-self-explaining-code/).
+
+Code comments are not a narrator's voice-over of the code, they should be little alerts.
+The more comments there are, the less likely it is that the reader will skip them.
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:121]: Is this correct?
```suggestion
The more comments there are, the more likely it is that the reader will skip them.
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+
+Code comments should not be a band-aid for bad naming or too complex code: instead of adding a comment, can you rename a variable or refactor a piece of code?
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:123]:
```suggestion
Code comments should not be a band-aid for bad naming or overly complex code: instead of adding a comment, can you rename a variable or refactor a piece of code?
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+
+An useful pattern is the idea of [explaining functions or variables](https://blog.r-hub.io/2023/01/26/code-comments-self-explaining-code/#use-helper-functions-or-explaining-variables) whereby code like
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:125]:
```suggestion
A useful idea is to use [self-explanatory functions or variables](https://blog.r-hub.io/2023/01/26/code-comments-self-explaining-code/#use-helper-functions-or-explaining-variables), where code like
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+Of course code comments remain important when needed!
+Examples of good comments include: 
+
+- function documentation with roxygen2, 
+- aspects you’d like to point out to a code reviewer, such as `# This query can not be done via GraphQL, so have to use v3 REST API`,
+- [comments that provide a table of contents](https://blog.r-hub.io/2023/01/26/code-comments-self-explaining-code/#use-comments-for-the-scripts-outline).
+
+## Clear code
+
+In the second part of the training, we shared tips that improve code's clarity.
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:151]:
```suggestion
In the second part of the training, we shared tips that improve code clarity.
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+### Self-explanatory names
+
+Naming things is notoriously hard.
+We shared these ideas:
+
+- Follow fashion, meaning, use the same words as others in your field or programming language.
+
+- Felienne Hermans, in her book [The Programmer's Brain](https://www.manning.com/books/the-programmers-brain), advises choosing the concepts that go into the name, the words to say it, then putting them together. This approach in three steps is a good way to get unstuck.
+
+- Following the previous advice, names should be consistent across code base and _name molds_ are a very good tool for that. Name molds are patterns in which the elements of a name are combined, for example if you calculate the maximum value of crop yield, you need to agree if `maximun` will be `max` or `maximun` and if the word will be at the beginning or at the end of the variable name: should be `maxYield` or `yieldMax`?  By normalizing how to name things, our code will be more easy to read.
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:162]:
```suggestion
- Following the previous advice, names should be consistent across code base and _name molds_ are a very good tool for that. Name molds are patterns in which the elements of a name are combined, for example if you calculate the maximum value of crop yield, you need to agree if `maximum` will be `max` or `maximum` and if the word will be at the beginning or at the end of the variable name: should be `maxYield` or `yieldMax`?  By normalizing how to name things, our code will be easier to read.     
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+Naming things is notoriously hard.
+We shared these ideas:
+
+- Follow fashion, meaning, use the same words as others in your field or programming language.
+
+- Felienne Hermans, in her book [The Programmer's Brain](https://www.manning.com/books/the-programmers-brain), advises choosing the concepts that go into the name, the words to say it, then putting them together. This approach in three steps is a good way to get unstuck.
+
+- Following the previous advice, names should be consistent across code base and _name molds_ are a very good tool for that. Name molds are patterns in which the elements of a name are combined, for example if you calculate the maximum value of crop yield, you need to agree if `maximun` will be `max` or `maximun` and if the word will be at the beginning or at the end of the variable name: should be `maxYield` or `yieldMax`?  By normalizing how to name things, our code will be more easy to read.     
+
+- “The greater the distance between a name’s declaration and its uses, the longer the name should be” (Andrew Gerrand). However, no matter, how close to defining a variable you use it, don't use a smart very short abbreviation.
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:164]:
```suggestion
- “The greater the distance between a name’s declaration and its uses, the longer the name should be” (Andrew Gerrand). However, no matter how close to defining a variable you use it, don't use a smart very short abbreviation.
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+
+- There are several way to write variables names, camelCase style lead to higher accuracy when reading code (Dave Binkley, 2009) and is better for reading the code with screen readers.  
+
+- A name is clear if the person reviewing your code agrees. 😉
+
+A further tip is that it's absolutely ok to create functions that wrap existing functions just to change their name.
+This strategy is common to change the argument order, but fine for naming too.
+Say you don't like seeing the French-looking "sans" instead of "without" in a function name, you can have:
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:172]: Just a suggestion?
```suggestion
Say you prefer your function names to be actions (verbs) rather than passive descriptions, you can have:
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+switch(type,
+  mean = mean(x),
+  median = median(x),
+  trimmed = mean(x, trim = .1)
+)
+```
+
+### Less code
+
+The code you don't write has no bug (that you are responsible for) and does not need to be read. :tada:
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:237]:
```suggestion
The code you *don't* write has no bug (that you are responsible for) and does not need to be read. :tada:
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 11]
```diff
@@ -0,0 +1,288 @@
+---
+title: Beautiful code, because we’re worth it!
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:2]:
```suggestion
title: Beautiful Code, Because We’re Worth It!
```

File: content/blog/2024-02-20-stylish-code/index.Rmd [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+author: 
+- Maëlle Salmon
+- Yanina Bellini Saibene
+date: '2024-02-22'
+slug: beautiful-code
+output: hugodown::md_document
+tags:
+  - champions program
+  - tech notes
```
Comment (steffilazerte) [inline content/blog/2024-02-20-stylish-code/index.Rmd:11]: Perhaps add some tags regarding the content of the post, rather than just why it was written? These tags already exist and I think are relevant.
```suggestion
  - tech notes
  - beginner
  - package development
  - clean code
```

File: content/blog/2024-02-20-stylish-code/index.Rmd (hunk 1 above)
Comment (steffilazerte) [reply content/blog/2024-02-20-stylish-code/index.Rmd:166]: Could also link to this cartoon by Allison Horst :grin: https://cdn.myportfolio.com/45214904-6a61-4e23-98d6-b140f8654a40/dbb99049-2916-4bc8-824f-1816f5c4f06d_rw_3840.png?h=5c28a07483856cd07de35a50c772824d
